kubernetes-client/src/main/java/io/fabric8/kubernetes/client/Config.java

```diff
@@ -537,8 +537,9 @@ private static boolean loadFromKubeconfig(Config config, String context, String
           config.setUsername(currentAuthInfo.getUsername());
           config.setPassword(currentAuthInfo.getPassword());
 
-          if (Utils.isNullOrEmpty(config.getOauthToken()) && currentAuthInfo.getAuthProvider() != null && !Utils.isNullOrEmpty(currentAuthInfo.getAuthProvider().getConfig().get(ACCESS_TOKEN))) {
-            config.setOauthToken(currentAuthInfo.getAuthProvider().getConfig().get(ACCESS_TOKEN));
+          if (Utils.isNullOrEmpty(config.getOauthToken()) && currentAuthInfo.getAuthProvider() != null && !Utils.isNullOrEmpty(currentAuthInfo.getAuthProvider().getConfig().get(
```

Comment:
I'm not really sure if this will introduce any regression as we are completely ditching `ACCES_TOKEN` which was added in #636 > #650

I would probably do as the issue suggests ("...Probably this code needs to be adapted to use either access-token or id-token, depending on what is available.."): so try to use ID_TOKEN or try to fallback to ACCESS_TOKEN

Reply:
@manusa good point... reading the `kubectl` source code I thought it cannot use `access-token` directly, but in fact it's used on gcloud.
I will refactor the code to support both cases.